src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,842 @@
+            {
+                return (AnalogInputMode)(GetRegisterBits(Register.Configuration, BitMask.ConfigurationPseudo));
+            }
+
+            set
+            {
+                SetRegisterBits(Register.Configuration, BitMask.ConfigurationPseudo, (uint)value);
+            }
+        }
+
+        /// <summary>
+        /// Changes the currently selected channel
+        /// </summary>
+        public Channel ActiveChannels
+        {
+            get
+            {
+                return (Channel)(GetRegisterBits(Register.Configuration, BitMask.ConfigurationCH));
+            }
+
+            private set
+            {
+                SetRegisterBits(Register.Configuration, BitMask.ConfigurationCH, (uint)value);
+            }
+        }
+
+        /// <summary>
+        /// Sets the amount of averaging. The data from the sinc filter is averaged by 2, 8, or 16. The averaging reduces the output data rate for a given FS word; however, the RMS noise improves.
+        /// </summary>
+        public AveragingMode Averaging
+        {
+            get
+            {
+                return (AveragingMode)(GetRegisterBits(Register.Mode, BitMask.ModeAVG));
+            }
+
+            set
+            {
+                SetRegisterBits(Register.Mode, BitMask.ModeAVG, (uint)value);
+            }
+        }
+
+        /// <summary>
+        /// Set the filter output data rate select bits. The 10 bits of data programmed into these bits determine the filter cutoff frequency, the position of the first notch of the filter, and the output data rate for the part.
+        /// </summary>
+        public ushort Filter
+        {
+            get
+            {
+                return (ushort)(GetRegisterBits(Register.Mode, BitMask.ModeFS));
+            }
+
+            set
+            {
+                SetRegisterBits(Register.Mode, BitMask.ModeFS, value);
+            }
+        }
+
+        /// <summary>
+        /// The value of the offset calibration register
+        /// </summary>
+        public uint Offset
+        {
+            get
+            {
+                return ReadRegisterValue(Register.Offset) & (uint)BitMask.Offset;
+            }
+
+            set
+            {
+                SetRegisterBits(Register.Offset, BitMask.Offset, value);
+            }
+        }
+
+        /// <summary>
+        /// The value of the full-scale calibration register
+        /// </summary>
+        public uint FullScale
+        {
+            get
+            {
+                return ReadRegisterValue(Register.FullScale) & (uint)BitMask.FullScale;
+            }
+
+            set
+            {
+                SetRegisterBits(Register.FullScale, BitMask.FullScale, value);
+            }
+        }
+
+        /// <summary>
+        /// Gets the metadata class for the device
+        /// </summary>
+        /// <returns></returns>
+        public static IDeviceMetadata GetDeviceMetadata()
+        {
+            return Ad7193.metadata;
```

Comment:
`metadata` should be Capitalized if it's public/protected or `_underscored` if it's private.